Sources/TuistLoader/Loaders/SwiftPackageManagerGraphLoader.swift

```diff
@@ -147,14 +151,22 @@ public final class SwiftPackageManagerGraphLoader: SwiftPackageManagerGraphLoadi
         )
 
         let externalProjects: [Path: ProjectDescription.Project] = try packageInfos.reduce(into: [:]) { result, packageInfo in
+            let path = packageInfo.folder
+            let packageType: PackageType
+            if path.components.contains(Constants.SwiftPackageManager.packageCheckoutDirectoryName) {
+                packageType = .remote(artifactPaths: packageToTargetsToArtifactPaths[packageInfo.name] ?? [:])
```

Comment:
Instead of assuming the checkouts path, what do you think about saving the `PackageType` in the `PackageInfo` when we initially load them from the `.build/workspace-state.json` file?

You can do that in the following switch: https://github.com/tuist/tuist/blob/main/Sources/TuistLoader/Loaders/SwiftPackageManagerGraphLoader.swift#L105-L118